entity-framework/core/miscellaneous/spatial.md

```diff
@@ -0,0 +1,295 @@
+Microsoft.EntityFrameworkCore.InMemory  | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite)
+Npgsql.EntityFrameworkCore.PostgreSQL   | [Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite)
+
+## Reverse engineering
+
+The spatial NuGet packages also enable [reverse engineering](../managing-schemas/scaffolding.md) models with spatial properties, but you need to install the package ***before*** running `Scaffold-DbContext` or `dotnet ef dbcontext scaffold`. If you don't, you'll receive warnings about not finding type mappings for the columns and the columns will be skipped.
+
+## NetTopologySuite (NTS)
+
+NetTopologySuite is a spatial library for .NET. EF Core enables mapping to spatial data types in the database by using NTS types in your model.
+
+To enable mapping to spatial types via NTS, call the UseNetTopologySuite method on the provider's DbContext options builder. For example, with SQL Server you'd call it like this.
+
+``` csharp
+optionsBuilder.UseSqlServer(
+    @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=WideWorldImporters",
+    x => x.UseNetTopologySuite());
+```
+
+There are several spatial data types. Which type you use depends on the types of shapes you want to allow. Here is the hierarchy of NTS types that you can use for properties in your model. They're located within the `NetTopologySuite.Geometries` namespace. Corresponding interfaces in the GeoAPI package can also be used.
+
+* Geometry
+  * Point
+  * LineString
+  * Polygon
+  * GeometryCollection
+    * MultiPoint
+    * MultiLineString
+    * MultiPolygon
+
+> [!WARNING]
+> CircularString, CompoundCurve, and CurePolygon aren't supported by NTS.
+
+The following entity classes could be used to map to tables in the [Wide World Importers sample database](http://go.microsoft.com/fwlink/?LinkID=800630).
+
+``` csharp
+[Table("Cities", Schema = "Application"))]
+class City
+{
+    public int CityID { get; set; }
+
+    public string CityName { get; set; }
+
+    public IPoint Location { get; set; }
+}
+
+[Table("Countries", Schema = "Application"))]
+class Country
+{
+    public int CountryID { get; set; }
+
+    public string CountryName { get; set; }
+
+    // Database includes both Polygon and MultiPolygon values
+    public IGeometry Border { get; set; }
+}
+```
+
+### Creating values
+
+You can use constructors to create geometry objects; however, NTS recommends using a geometry factory instead. This lets you specify a default SRID and gives you control over more advanced things like the precision model (used during calculations) and the coordinate sequence (determines which ordinates--dimensions and measures--are available).
+
+``` csharp
+var geometryFactory = NtsGeometryServices.Instance.CreateGeometryFactory(srid: 4326);
```

Comment:
Maybe add a note about 4326 and some wikipedia(?) link to coordinate systems?

Reply:
BTW, I understand we may want to stay away from trying to explain a lot of these things in our own docs but I think there is a set of customers for which we are making spatial data more accessible, and they may find links and brief explanations useful.